cw_solana/lib/solana_wallet.dart

```diff
@@ -0,0 +1,460 @@
+    required String password,
+    SolanaBalance? initialBalance,
+  })  : syncStatus = const NotConnectedSyncStatus(),
+        _password = password,
+        _mnemonic = mnemonic,
+        _hexPrivateKey = privateKey,
+        _isTransactionUpdating = false,
+        _client = SolanaWalletClient(),
+        walletAddresses = SolanaWalletAddresses(walletInfo),
+        balance = ObservableMap<CryptoCurrency, SolanaBalance>.of(
+            {CryptoCurrency.sol: initialBalance ?? SolanaBalance(BigInt.zero.toDouble())}),
+        super(walletInfo) {
+    this.walletInfo = walletInfo;
+    transactionHistory = SolanaTransactionHistory(walletInfo: walletInfo, password: password);
+
+    if (!CakeHive.isAdapterRegistered(SPLToken.typeId)) {
+      CakeHive.registerAdapter(SPLTokenAdapter());
+    }
+
+    _sharedPrefs.complete(SharedPreferences.getInstance());
+  }
+
+  final String _password;
+  final String? _mnemonic;
+  final String? _hexPrivateKey;
+
+  // The Solana WalletPair
+  Ed25519HDKeyPair? _wallet;
+
+  // To access the privateKey bytes.
+  Ed25519HDKeyPairData? _keyPair;
+
+  late SolanaWalletClient _client;
+
+  bool _isTransactionUpdating;
+
+  Timer? _transactionsUpdateTimer;
+
+  late final Box<SPLToken> splTokensBox;
+
+  @override
+  WalletAddresses walletAddresses;
+
+  @override
+  @observable
+  SyncStatus syncStatus;
+
+  @override
+  @observable
+  late ObservableMap<CryptoCurrency, SolanaBalance> balance;
+
+  Completer<SharedPreferences> _sharedPrefs = Completer();
+
+  @override
+  Ed25519HDKeyPairData get keys {
+    if (_keyPair == null) {
+      return Ed25519HDKeyPairData([], publicKey: const Ed25519HDPublicKey([]));
+    }
+
+    return _keyPair!;
+  }
+
+  @override
+  String? get seed => _mnemonic;
+
+  @override
+  String get privateKey => HEX.encode(_keyPair!.bytes);
+
+  Future<void> init() async {
+    final boxName = "${walletInfo.name.replaceAll(" ", "_")}_${SPLToken.boxName}";
+
+    splTokensBox = await CakeHive.openBox<SPLToken>(boxName);
+
+    // Create WalletPair using either the mnemonic or the privateKey
+    _wallet = await getWalletPair(
+      mnemonic: _mnemonic,
+      privateKey: _hexPrivateKey,
+    );
+
+    // Extract the keyPairData containing both the privateKey bytes and the publicKey hex.
+    _keyPair = await _wallet!.extract();
+
+    walletInfo.address = _wallet!.address;
+
+    print(_wallet!.address);
+
+    await walletAddresses.init();
+    await transactionHistory.init();
+    await save();
+  }
+
+  Future<Wallet> getWalletPair({String? mnemonic, String? privateKey}) async {
+    assert(mnemonic != null || privateKey != null);
+
+    if (privateKey != null) {
+      final privateKeyBytes = hexToBytes(privateKey);
+      return await Wallet.fromPrivateKeyBytes(privateKey: privateKeyBytes);
+    }
+
+    return Wallet.fromMnemonic(mnemonic!, account: 0, change: 0);
+  }
+
+  @override
+  int calculateEstimatedFee(TransactionPriority priority, int? amount) => 0;
```

Comment:
This has to do with the transaction priority, and as at last discussion, we mentioned putting it on hold for now because the libraries we use currently do not provide a clear cut way for specifying the priority fee.